modules/cnv_upload_virtctl.adoc

```diff
@@ -1,51 +1,50 @@
 [[upload-vmdisk-virtctl]]
 === Uploading a local disk image to a new PVC
 
-You can use *virtctl image-upload* to upload a virtual machine disk image from 
-a client machine to your {product-title} cluster. This creates a PVC that can be 
+You can use `virtctl image-upload` to upload a virtual machine disk image from
+a client machine to your {product-title} cluster. This creates a PVC that can be
 associated with a virtual machine after the upload has completed.
 
 .Prerequisites
 
-* A virtual machine disk image, in RAW or QCOW2 format. It can be compressed 
+* A virtual machine disk image, in RAW or QCOW2 format. It can be compressed
 using *xz* or *gzip*.
 * *kubevirt-virtctl* must be installed on the client machine.
+* The client machine must be xref:../install_config/router/default_haproxy_router.html#overview[configured]
+to trust the OpenShift router's certificate.
 
 .Procedure
 
 . Identify the following items:
 * File location of the VM disk image you want to upload
 * Name and size desired for the resulting PVC
- 
-. Expose the *cdi-uploadproxy* service so that you can upload data to your cluster:
+
+. Remove the existing passthrough route:
++
+----
+$ oc delete route -n cdi cdi-uploadproxy-route
+----
+
+. Create a secured route using re-encryption termination:
 +
 ----
-cat <<EOF | oc apply -f -
-apiVersion: v1
-kind: Route
-metadata:
-  name: cdi-uploadproxy
-  namespace: kube-system 
-spec:
-  to:
-    kind: Service
-    name: cdi-uploadproxy 
-  tls:
-    termination: passthrough
-EOF
-----
-
-. Use the `virtctl image-upload` command to upload your VM image, 
+$ oc get secret -n cdi cdi-upload-proxy-ca-key -o=jsonpath="{.data['tls\.crt']}" | base64 -d > ca.pem
+
+$ oc create route reencrypt -n cdi --service=cdi-uploadproxy --dest-ca-cert=ca.pem
```

Comment:
please specify the route name changing the above line to:

oc create route reencrypt cdi-uploadproxy-route -n cdi --service=cdi-uploadproxy --dest-ca-cert=ca.pem

Reply:
Done in the latest commit. Preview here: http://file.bos.redhat.com/pousley/022519/route/cnv_users_guide/cnv_users_guide.html#upload-vmdisk-virtctl